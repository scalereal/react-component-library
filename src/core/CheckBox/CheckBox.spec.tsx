import { render } from '@testing-library/react';
import * as React from 'react';
import { CheckBox } from '.';

describe('CheckBox component', () => {
    it('should render with default theme', () => {
        const { container } = render(<CheckBox/>);
        expect(container).toMatchSnapshot();
    });

    it('should render with all the props given', () => {
        const { container } = render(
            <CheckBox
                label="1"
                value={1}
                size="M"
                fontSize='M'
                labelColor='white'
                bgColor= 'blue'
                hoverEffect={true}
                labelHoverEffect={true}
                hoverBgColor='#126AFA'
                labelHoverColor= '#12fadb'
                hoverShadowColor="rgba(96, 120, 144, 0.35)"
            />
            );
        expect(container).toMatchSnapshot();
    });

    it('should render with not-allowed cursor when it is disabled', () => {
        const { container } = render(
            <CheckBox disabled={true} />
        );
        expect(container).toMatchSnapshot();
    });

    // it('should render with correct children', () => {
    //     const{container}=render(<CheckBox label="1" value={1} />);
    //     expect(container).toContainHTML("<input type='checkbox' value='1' /><span>1</span>");
    // });

});