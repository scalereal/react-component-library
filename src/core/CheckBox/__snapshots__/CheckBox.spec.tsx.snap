// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox component should render with all the props given 1`] = `
<div>
  <div
    class="sc-hKgILt cTJgxF"
    font-size="M"
    label="1"
    value="1"
  >
    <label
      class="sc-gsTCUz jbqcjW"
      for="error"
      id="error"
    />
    <label
      class="sc-dlfnbm UwbaL"
      font-size="M"
      label="1"
      value="1"
    >
      <input
        type="checkbox"
        value="1"
      />
      <span
        class="sc-bdfBwQ gdLRZY"
      >
        1
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBox component should render with default theme 1`] = `
<div>
  <div
    class="sc-hKgILt cTJgxF"
  >
    <label
      class="sc-gsTCUz jbqcjW"
      for="error"
      id="error"
    />
    <label
      class="sc-dlfnbm knavht"
    >
      <input
        type="checkbox"
        value="CheckBox"
      />
      <span
        class="sc-bdfBwQ cmSgzP"
      >
        CheckBox
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBox component should render with not-allowed cursor when it is disabled 1`] = `
<div>
  <div
    class="sc-hKgILt cTJgxF"
    disabled=""
  >
    <label
      class="sc-gsTCUz jbqcjW"
      for="error"
      id="error"
    />
    <label
      class="sc-dlfnbm kVdZtZ"
      disabled=""
    >
      <input
        type="checkbox"
        value="CheckBox"
      />
      <span
        class="sc-bdfBwQ cmSgzP"
      >
        CheckBox
      </span>
    </label>
  </div>
</div>
`;
